src/fiber.cr

```diff
@@ -46,9 +45,21 @@ class Fiber
 
     fiber_main = ->(f : Fiber) { f.run }
 
-    # point to first addressable pointer on the stack (@stack_bottom points past
-    # the stack because the stack grows down):
-    stack_ptr = @stack_bottom - sizeof(Void*)
+    # FIXME: This line shouldn't be necessary (#7975)
```

Comment:
One way to remove this FIXME is to do the if flag check twice, but the second one inside the assignment to stack_ptr. But it's not necessary, I think this way is fine too.